crates/iota/src/upgrade_compatibility.rs

```diff
@@ -658,14 +725,686 @@ fn missing_definition_diag(
         std::iter::empty::<(Loc, String)>(),
         vec![
             format!(
-                "{declaration_kind} is missing expected {declaration_kind} '{identifier_name}', but found none",
+                "{declaration_kind} is missing expected {declaration_kind} '{identifier_name}', \
+                but found none",
             ),
             format!(
-                "{declaration_kind}s are part of a module's public interface and cannot be removed or changed during an upgrade",
+                "{declaration_kind}s are part of a module's public interface \
+                     and cannot be removed or changed during an upgrade.",
             ),
             format!(
-                "add missing {declaration_kind} '{identifier_name}' back to the module '{module_name}'.",
+                "add missing {declaration_kind} '{identifier_name}' \
+                     back to the module '{module_name}'.",
             ),
         ],
-    )
+    ));
+
+    Ok(diags)
+}
+
+/// Return a diagnostic for a function signature mismatch.
+/// start by checking the lengths of the parameters and returns and return a
+/// diagnostic if they are different if the lengths are the same check each
+/// parameter piece wise and return a diagnostic for each mismatch
```

Comment:
```suggestion
/// Return a diagnostic for a function signature mismatch.
/// 
/// A function signature does not match if the number of parameters is different, or if corresponding parameters are different.
```

Reply:
@DaughterOfMars feel free to edit this suggestion if you prefer something else.

Reply:
Looks fine but I bet it will need to be formatted